server/service/scripts.go

```diff
@@ -192,26 +201,25 @@ func (svc *Service) RunHostScript(ctx context.Context, request *fleet.HostScript
 		return nil, fleet.NewInvalidArgumentError("script_contents", err.Error())
 	}
 
-	// host must be online
-	if host.Status(time.Now()) != fleet.StatusOnline {
+	asyncExecution := waitForResult <= 0
+
+	if !asyncExecution && host.Status(time.Now()) != fleet.StatusOnline {
 		return nil, fleet.NewInvalidArgumentError("host_id", fleet.RunScriptHostOfflineErrMsg)
 	}
 
-	// TODO: to be adjusted, we don't check the pending scripts anymore unless it is a synchronous request.
-	//// it is important that the "ignoreOlder" parameter in this call is the same
-	//// everywhere (which is here and in the "get orbit config" endpoint to send
-	//// the notification of scripts pending execution to the host).
-	//pending, err := svc.ds.ListPendingHostScriptExecutions(ctx, request.HostID, scripts.MaxServerWaitTime)
-	//if err != nil {
-	//	return nil, ctxerr.Wrap(ctx, err, "list host pending script executions")
-	//}
-	//if len(pending) > 0 {
-	//	return nil, fleet.NewInvalidArgumentError(
-	//		"script_contents", fleet.RunScriptAlreadyRunningErrMsg,
-	//	).WithStatus(http.StatusConflict)
-	//}
+	pending, err := svc.ds.ListPendingHostScriptExecutions(ctx, request.HostID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "list host pending script executions")
+	}
+	if len(pending) > 1000 {
```

Comment:
Let's make that `1000` a `const` with a clear name, and if needed we can add a `var testMaxPendingScripts int` that can be set in tests and override the `const` (that is, inside this func we can do:

```
maxPending := maxPendingScripts // i.e. local var = package const
if testMaxPendingScripts > 0 {
  maxPending = testMaxPendingScripts // overridable for tests
}
// and then use maxPending inside the func to do the actual validation
```